src/functions/supsub.js

```diff
@@ -0,0 +1,239 @@
+        // Operators handle supsubs differently when they have limits
+        // (e.g. `\displaystyle\sum_2^3`)
+        const delegate = base.value.limits &&
+            (options.style.size === Style.DISPLAY.size ||
+            base.value.alwaysHandleSupSub);
+        return delegate ? op.htmlBuilder : null;
+    } else if (base.type === "accent") {
+        return utils.isCharacterBox(base.value.base) ? accent.htmlBuilder : null;
+    } else if (base.type === "horizBrace") {
+        const isSup = !group.value.sub;
+        return isSup === base.value.isOver ? horizBrace.htmlBuilder : null;
+    } else {
+        return null;
+    }
+};
+
+// Super scripts and subscripts, whose precise placement can depend on other
+// functions that precede them.
+defineFunctionBuilders({
+    type: "supsub",
+    htmlBuilder(group, options) {
+        // Superscript and subscripts are handled in the TeXbook on page
+        // 445-446, rules 18(a-f).
+
+        // Here is where we defer to the inner group if it should handle
+        // superscripts and subscripts itself.
+        const builderDelegate = htmlBuilderDelegate(group, options);
+        if (builderDelegate) {
+            return builderDelegate(group, options);
+        }
+
+        const {base: valueBase, sup: valueSup, sub: valueSub} = group.value;
+        const base = html.buildGroup(valueBase, options);
+        let supm;
+        let subm;
+
+        const metrics = options.fontMetrics();
+
+        // Rule 18a
+        let supShift = 0;
+        let subShift = 0;
+
+        const isCharacterBox = valueBase && !utils.isCharacterBox(valueBase);
+        if (valueSup) {
+            const newOptions = options.havingStyle(options.style.sup());
+            supm = html.buildGroup(valueSup, newOptions, options);
+            if (!isCharacterBox) {
+                supShift = base.height - newOptions.fontMetrics().supDrop
+                    * newOptions.sizeMultiplier / options.sizeMultiplier;
+            }
+        }
+
+        if (valueSub) {
+            const newOptions = options.havingStyle(options.style.sub());
+            subm = html.buildGroup(valueSub, newOptions, options);
+            if (!isCharacterBox) {
+                subShift = base.depth + newOptions.fontMetrics().subDrop
+                    * newOptions.sizeMultiplier / options.sizeMultiplier;
+            }
+        }
+
+        // Rule 18c
+        let minSupShift;
+        if (options.style === Style.DISPLAY) {
+            minSupShift = metrics.sup1;
+        } else if (options.style.cramped) {
+            minSupShift = metrics.sup3;
+        } else {
+            minSupShift = metrics.sup2;
+        }
+
+        // scriptspace is a font-size-independent size, so scale it
+        // appropriately for use as the marginRight.
+        const multiplier = options.sizeMultiplier;
+        const marginRight = (0.5 / metrics.ptPerEm) / multiplier + "em";
+
+        let supsub;
+        if (supm && subm) {
```

Comment:
The original code structure was:
```
if (!group.value.sup) {
  ...
} else if (!group.value.sub) {
  ...
} else {
  ...
}
```
As in many other refactors, this was reordered to make positive/existence checks (which requires start with the original `else` case where both are non-nullable) instead of negative/non-existence checks so that non-nullability of the `supm` and `subm` are immediately obvious to the type system.